### PR TITLE
Handle cases where guessit returns a list for a show name

### PR DESCRIPTION
**Pull request by offbyone**
Works around the impact of https://github.com/guessit-io/guessit/issues/770

Proposed changes in this pull request:
- Add some tests for the sanitize name functions based on current behaviour
- When `full_sanitizeSceneName` receives an iterable that is not a string, turn it into one first.
- A tiny tweak to the pyproject.toml so that [ruff](https://docs.astral.sh/ruff/) will respect the project line length too

- [X] PR is based on the DEVELOP branch
- [X] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [X] Read [contribution guide](https://github.com/SickChill/SickChill/blob/master/.github/CONTRIBUTING.md)

This is the log message / traceback I'm addressing:

```
AA AttributeError: 'list' object has no attribute 'replace'
AA ^^^^^^^^^^^^
AA name = name.replace(x, "")
AA File "/app/sickchill/lib/python3.11/site-packages/sickchill/oldbeard/helpers.py", line 668, in sanitizeSceneName
AA ^^^^^^^^^^^^^^^^^^^^^^^
AA return re.sub("[. -]", " ", sanitizeSceneName(name)).lower().strip()
AA File "/app/sickchill/lib/python3.11/site-packages/sickchill/oldbeard/helpers.py", line 855, in full_sanitizeSceneName
AA ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AA name = helpers.full_sanitizeSceneName(name)
AA File "/app/sickchill/lib/python3.11/site-packages/sickchill/oldbeard/name_cache.py", line 37, in get_id_from_name
AA ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AA cache = sickchill.oldbeard.name_cache.get_id_from_name(name)
AA File "/app/sickchill/lib/python3.11/site-packages/sickchill/oldbeard/helpers.py", line 885, in get_show
```

## Summary by CodeRabbit

- **New Features**
	- Enhanced the app's ability to process non-string iterable inputs in scene name sanitization.
- **Refactor**
	- Improved organization and import structure in helper functions.
- **Tests**
	- Removed skip decorators on certain tests, enabling them to run.
	- Added new test cases for scene name sanitization functions.
- **Chores**
	- Configured code styling tool with updated settings.

### CODE REVIEW

File: sickchill/oldbeard/helpers.py [hunk 1]
```diff
@@ -35,8 +36,8 @@
 
 import sickchill
 from sickchill import adba, logger, settings
-from sickchill.helper import episode_num, pretty_file_size, SUBTITLE_EXTENSIONS
-from sickchill.helper.common import is_media_file, replace_extension, USER_AGENT
+from sickchill.helper import SUBTITLE_EXTENSIONS, episode_num, pretty_file_size
+from sickchill.helper.common import USER_AGENT, is_media_file, replace_extension
```
Comment (offbyone) [inline sickchill/oldbeard/helpers.py]: ruff is really aggressive in re-ordering imports; I hope this is okay, if not, I can _try_ make it stop :D

File: sickchill/oldbeard/helpers.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import urllib.request
 import uuid
 import zipfile
+from collections.abc import Iterable
```
Comment (miigotu) [inline sickchill/oldbeard/helpers.py:19]: You have to try and except here and try collections.abc.Iterable and collections.iterable. The aliases moved in python 3.9 -> 3.10.

**Comment by miigotu** [comment]
Also, the fake_is_active fix is in a comment I made in the other PR @BKSteve

**Comment by miigotu** [comment]
Replaced by #8759

**Comment by miigotu** [comment]
@offbyone if you want to add those tests to `test_guessit_findit` that would be dope

**Pull request closed without merging**